### DISCUSSION TITLE
Possibility to run on local git repo

### DISCUSSION THREAD

**Discussion by containerpope** [fluxcd/flux2#2031] [Q&A] [answered]
I have a case, where I run a single node k3s cluster on the edge. The updates come from the cloud through a separate software. I thought about putting all the manifests into a local git repo and committing updates to it, so that I could run the flux source and kustomize controllers to watch and apply. This would allow me to use dependsOn and other cool features like garbage collection. The source controller would then mount the repo locally into the pod.
Is there any way I could make this (or an alternative) setup work, maybe through some local testing flags? Thank you in advance for any help on this 🙂

**Comment by stefanprodan** [comment, marked as answer]
You could use a minimal Git server based on Alpine and deploy it next to Flux, e.g. https://github.com/fluxcd/gitsrv

**Reply by containerpope** [reply]
@stefanprodan thank you a lot for the hint. That should do the job without code changes. 

How much effort would you see in extending the source-controller with a local repository type? In that case we would mount the repository via a local directory. The other logic that is periodically checking the repo would stay the same. 
This would allow us to reduce components involved that can fail and unnecessary network traffic. It is just a theoretical question, but any input is highly welcome 😊

**Reply by stefanprodan** [reply]
I'm not for adding such a feature to source-controller, the controller sole role is to fetch resources from external services, hosting Git repos is out of scope IMO.

**Reply by wibed** [reply]
@stefanprodan

its not meant to host a git repo, but to refer to a bare git repo by path as in `git init -bare`
this would allow for fast spin up locally.
